crates/sui-indexer/migrations/pg/2024-09-25-211901_drop_network_total_transactions/up.sql

```diff
@@ -0,0 +1,2 @@
+ALTER TABLE checkpoints DROP COLUMN network_total_transactions;
+UPDATE checkpoints SET max_tx_sequence_number = max_tx_sequence_number + 1 WHERE max_tx_sequence_number IS NOT NULL;
```

Comment:
I don't actually understand. Why + 1?

Reply:
Ashok proposed a slightly different semantics for min and max seq numbers as

> There is an alternative, which is to store (for checkpoint C):
tx_lo = 1 + max(tx for checkpoint c < C for tx in c)
tx_hi = 1 + max(tx for checkpoint c <= C for tx in c) 
These are guaranteed to exist, even for empty checkpoints, so we can use a NOT NULL constraint if we wanted to.
You can compute network_total_transactions from them (it's tx_hi if transactions are 0-indexed, or tx_hi - 1 otherwise).
The number of transactions in C is always tx_hi - tx_lo.

Basically that changes the max_tx_sequence_number from inclusive to exclusive, hence the +1 for the semantics change.

Reply:
Are there empty checkpoints?
If we do want to go that route, I think max_tx_sequence_number would be a very confusing name.

Reply:
It seems `network_total_transactions` would have a much clearer semantics than the new `max_tx_sequence_number`

Reply:
> Are there empty checkpoints?

no empty checkpoint yet.

> If we do want to go that route, I think max_tx_sequence_number would be a very confusing name

this sounds familiar :) -- I actually proposed `network_total_transactions` & `checkpoint_total_transaction` instead of min & `max_tx_sequence_number` in a thread above with @amnn, I think that's more intuitive indeed https://github.com/MystenLabs/sui/pull/19560#discussion_r1776856973 I saw Ashok's broader point about pre-computation and filtering too though hence changing it that way

Reply:
hmm min&max <> lo&hi are sorta the same to me, the names per se do not indicate they are inclusive or exclusive -- it's actually good if tx_hi is less confusing and happy to do it that way

Reply:
There is a difference: `min` and `max` are definitely inclusive bounds, whereas `lo` and `hi` do not have that association, so people are more likely to read the docs to confirm whether the bound is exclusive or not and less likely to be confused by an exclusive `hi` vs an exclusive `max`. What's more, these terms are already used this way in the GraphQL codebase that will read these tables so there will also be no confusion there (and we document the convention there too).